precompiles/Modexp.yul

```diff
@@ -333,6 +341,41 @@ object "ModExp" {
 
             }
 
+            /// @notice Performs the multiplication between two bigUInts
+            /// @dev The result is stored from `mulResultPtr` to `mulResultPtr + (LIMB_SIZE * nLimbs)`.
```

Comment:
It would be nice to have a reference to a book, paper or another repository if we're drawing inspiration or imitating a certain implementation of Big Integer Multiplication, for future reference and bug fixing.